### ISSUE TITLE
Navigation is delayed sometimes but mostly not moving

### ISSUE THREAD

**Issue by XOR0831** [ppizarror/pygame-menu#359]
**Environment information**
Describe your environment information, such as:
- SO: linux
- python version: v3.7.3
- pygame version: v1.9.6
- pygame-menu version: v4.1.1

**Describe the bug**
The navigation to Menu options is delayed or not moving sometimes. I am using PyAutoGUI to emulate some keyboard press event from GPIO. When a GPIO pin is HIGH it calls the function press from PyAutoGUI. Also, Im trying to map WASD button to arrow keys. The keyboard event is catched by the pygame.events but the selector is sometimes moves but mostly it doesn't move.

**To Reproduce**

    from pyautogui import press

    while True:
        main_menu = pygame_menu.Menu(
            title="Main Test Menu",
            width=800,
            height=600,
            theme=pygame_menu.themes.THEME_DARK,
            joystick_enabled=False,
            mouse_enabled=False,
            mouse_motion_selection=False,
            touchscreen=False,
            touchscreen_motion_selection=False,
            mouse_visible=False,
        )
    
        main_menu.add.button("Input Test", input_menu)
        main_menu.add.button("Steering Wheel Test", steer_test_menu)
        main_menu.add.button("Motor Test", motor_menu)
        main_menu.add.button("Coin Chute Test", coin_menu)
        main_menu.add.button("Time Settings", time_menu)

        if button1.is_pressed:
            press("up")
        elif button2.is_pressed:
            press("down")
        elif button3.is_pressed:
            press("left")
        elif button4.is_pressed:
            press("right")
    
        events = pygame.event.get()
        for event in events:
            if event.type == pygame.QUIT:
                gameExit = True
    
            if event.type == pygame.KEYDOWN:
                if event.key == pygame.K_LEFT:
                    print("Left")
                elif event.key == pygame.K_RIGHT:
                    print("Right")
                elif event.key == pygame.K_UP:
                    print("Up")
                elif event.key == pygame.K_DOWN:
                    print("Down")
                elif event.key == pygame.K_d:
                    press("right")
                elif event.key == pygame.K_a:
                    press("left")
                elif event.key == pygame.K_w:
                    press("up")
                elif event.key == pygame.K_s:
                    press("down")
    
        if main_menu.is_enabled():
            main_menu.update(events)
            main_menu.draw(gameDisplay)
    
        pygame.display.flip()

**Comment by XOR0831** [comment]
The remapping issue was solved on #358 but the delayed and not working/updating navigation still needs to be addressed.

**Comment by XOR0831** [comment]
You may try to use press("down") / press("up") function from pyautogui to check the issue. You will see there that even there is a keyboard event that is received on pygame events the navigation does not move down or up.

**Comment by XOR0831** [comment]
**Simplifying code to reproduce the issue**

    import pygame
    import pygame_menu
    from pyautogui import press
    pygame.init()
    
    gameDisplay = pygame.display.set_mode((800, 600))
    while True:
        main_menu = pygame_menu.Menu(
            title="Main Test Menu",
            width=800,
            height=600,
            theme=pygame_menu.themes.THEME_DARK,
            joystick_enabled=False,
            mouse_enabled=False,
            mouse_motion_selection=False,
            touchscreen=False,
            touchscreen_motion_selection=False,
            mouse_visible=False,
        )
    
        main_menu.add.button("Input Test", None)
        main_menu.add.button("Steering Wheel Test", None)
        main_menu.add.button("Motor Test", None)
        main_menu.add.button("Coin Chute Test", None)
        main_menu.add.button("Time Settings", None)
    
        press("down") # This emulates the arrow down key being pressed
    
        events = pygame.event.get() # The press down is captured here
        for event in events:
            if event.type == pygame.QUIT:
                gameExit = True
    
            if event.type == pygame.KEYDOWN:
                if event.key == pygame.K_LEFT:
                    print("Left")
                elif event.key == pygame.K_RIGHT:
                    print("Right")
                elif event.key == pygame.K_UP:
                    print("Up")
                elif event.key == pygame.K_DOWN:
                    print("Down")
    
        if main_menu.is_enabled():
            main_menu.update(events)
            main_menu.draw(gameDisplay)
    
        pygame.display.flip()

**Comment by ppizarror** [comment]
Hi! I'll look into this issue as soon as I can

**Comment by ppizarror** [comment]
I know why this feature does not work, the Menu and widgets ignore invalid keyboard events if they're not pressed, https://github.com/ppizarror/pygame-menu/blob/cde28c272d79b1d12c9f2fda9fcf92800d883b73/pygame_menu/menu.py#L2357-L2359

That function checks if the key is pressed (using pygame), https://github.com/ppizarror/pygame-menu/blob/cde28c272d79b1d12c9f2fda9fcf92800d883b73/pygame_menu/utils.py#L240-L256

This has been done to prevent repeating issues due to textinput, gamepads, etc. Is there a way to check that the event is virtual (basically, to know if pygame.key.get_pressed() is True for such event)?

If that line is removed, the menu works as intended

**Comment by ppizarror** [comment]
Also, the example has a minor issue:

```python
gameDisplay = pygame.display.set_mode((800, 600))
main_menu = pygame_menu.Menu(
    title="Main Test Menu",
    width=800,
    height=600,
    theme=pygame_menu.themes.THEME_DARK,
    joystick_enabled=False,
    mouse_enabled=False,
    mouse_motion_selection=False,
    touchscreen=False,
    touchscreen_motion_selection=False,
    mouse_visible=False,
)

main_menu.add.button("Input Test", None)
main_menu.add.button("Steering Wheel Test", None)
main_menu.add.button("Motor Test", None)
main_menu.add.button("Coin Chute Test", None)
main_menu.add.button("Time Settings", None)

while True:

    press("down")  # This emulates the arrow down key being pressed

    events = pygame.event.get()  # The press down is captured here
    for event in events:
        if event.type == pygame.QUIT:
            gameExit = True

        if event.type == pygame.KEYDOWN:
            if event.key == pygame.K_LEFT:
                print("Left")
            elif event.key == pygame.K_RIGHT:
                print("Right")
            elif event.key == pygame.K_UP:
                print("Up")
            elif event.key == pygame.K_DOWN:
                print("Down")  # This will be printed but the navigation is not moving down

    if main_menu.is_enabled():
        main_menu.update(events)
        main_menu.draw(gameDisplay)

    pygame.display.flip()
```

**Comment by ppizarror** [comment]
Anyway, If there's no other elegant solution I can add a property to the Menu:

```python
main_menu._ignore_physical_keydown = False # True by default
```

If ``False`` the condition is not checked and the events should work as intended

**Comment by XOR0831** [comment]
Yes please, can you add that?

**Comment by ppizarror** [comment]
I've fixed this issue in #360, sadly it cannot be tested as travis.ci does not offers a DISPLAY to tests the feature (https://github.com/asweigart/pyautogui/issues/585, https://github.com/asweigart/pyautogui/issues/549, https://github.com/asweigart/pyautogui/issues/161). However, locally it works.

See ``test_pyautogui`` method in https://github.com/ppizarror/pygame-menu/pull/360/files#diff-adf98c23f577df905efd82739f8add32de8903b21b15b1783513b23558d0848c

Let me know if the PR solves your issue. If so, I'll push the new version 4.1.2 to PyPI 🍔

**Comment by XOR0831** [comment]
It works! Thank you so much!
